### ISSUE TITLE
A strange case in Bisection

### ISSUE THREAD

**Issue by HajimeKawahara** [google/jaxopt#191]
I just wanted to report that I came across a strange case while testing Bisection:

``` python
def F(x):
    return x-0.5001
bisec = Bisection(optimality_fun=F, lower=0.0, upper=1.0)
print(bisec.run().params)
```
this correctly provides 
```sh
0.5001068
```
But when I use the exact 0.5 as the solution 
``` python
def F(x):
    return x-0.5
bisec = Bisection(optimality_fun=F, lower=0.0, upper=1.0)
print(bisec.run().params) 
```
I got 
```sh
0.75
```
So, my assumption is that this occurs only when the solution is (lower+upper)/2. The other test supports this assumption:

 ``` python
val0=[]
x0arr=np.linspace(0.4,0.6,11)
epsilon=0.0 
for x0 in x0arr:
    def F(x):
        return x-x0+epsilon
    bisec = Bisection(optimality_fun=F, lower=0.0, upper=2*x0)
    val0.append(bisec.run().params)
np.array(val0)
```

epsilon=0 and 0.0001 gives

![test](https://user-images.githubusercontent.com/15956904/160260439-fee9a2bf-7afc-4f01-9a28-f6e8555f3d69.png)

BTW, can I include` Bisection` in `jax.lax.scan`?

Thanks a lot.

**Comment by mblondel** [comment]
Thanks a lot for the bug report! The issue should be fixed in #192.

> BTW, can I include Bisection in jax.lax.scan?

Bisection is jittable if you set `check_bracket=False`. When `check_bracket=True`, we check if the function values at `lower` and `upper` have opposite signs but this is not a jittable check.

**Comment by HajimeKawahara** [comment]
I see. Thank you!
